tests/integration/test_reactive.py

```diff
@@ -4,137 +4,227 @@
 """Testing reactive mode. This is only supported for the OpenStack cloud."""
 import json
 import re
+from typing import AsyncIterator
 
 import pytest
+import pytest_asyncio
 from github import Branch, Repository
+from github.WorkflowRun import WorkflowRun
 from github_runner_manager.reactive.consumer import JobDetails
 from juju.application import Application
 from juju.model import Model
 from juju.unit import Unit
 from kombu import Connection
 from pytest_operator.plugin import OpsTest
 
+from charm_state import VIRTUAL_MACHINES_CONFIG_NAME
 from tests.integration.helpers.common import (
+    DISPATCH_CRASH_TEST_WORKFLOW_FILENAME,
     DISPATCH_TEST_WORKFLOW_FILENAME,
     dispatch_workflow,
     reconcile,
+    wait_for,
     wait_for_completion,
+    wait_for_status,
 )
 
 pytestmark = pytest.mark.openstack
 
 
-@pytest.fixture(name="setup_queue", autouse=True)
+@pytest_asyncio.fixture(name="app")
+async def app_fixture(app_for_reactive: Application) -> AsyncIterator[Application]:
+    """Setup the reactive charm with 1 virtual machine and tear down afterwards."""
+    await app_for_reactive.set_config({VIRTUAL_MACHINES_CONFIG_NAME: "1"})
+    await reconcile(app_for_reactive, app_for_reactive.model)
+
+    yield app_for_reactive
+
+    # Call reconcile to enable cleanup of any runner spawned
+    await app_for_reactive.set_config({VIRTUAL_MACHINES_CONFIG_NAME: "0"})
+    await reconcile(app_for_reactive, app_for_reactive.model)
+
+
+@pytest_asyncio.fixture(name="setup_queue", autouse=True)
 async def setup_queue_fixture(
     ops_test: OpsTest,
-    app_for_reactive: Application,
+    app: Application,
 ):
-    mongodb_uri = await _get_mongodb_uri(ops_test, app_for_reactive)
+    mongodb_uri = await _get_mongodb_uri(ops_test, app)
 
-    _clear_queue(mongodb_uri, app_for_reactive.name)
-    _assert_queue_is_empty(mongodb_uri, app_for_reactive.name)
+    _clear_queue(mongodb_uri, app.name)
+    _assert_queue_is_empty(mongodb_uri, app.name)
 
 
+@pytest.mark.abort_on_fail
```

Comment:
Let the other test fail because the tests are not independent from each other. If spawning a runner fails, it seems to affect the other tests.

Reply:
In what way do they depend on each other? Is it that if this test fails, the other tests are likely to fail or do the other tests use some part of the setup of this test?

Reply:
They shouldn't logically depend on each other, but it seemed that the setup/teardown code wasn't as effective as I wanted it to be at isolating these tests. I changed it to clear the queue in the application setup before reconciliation. I also removed the abort_on_fail.